app/src/main/java/org/liberty/android/fantastischmemo/utils/HintUtil.java

```diff
@@ -14,20 +16,34 @@
     public HintUtil() {
     }
 
-    public String generateLetterHint(int count, String word) {
+    public static String generateLetterHint(int count, String word, HashSet<Integer> used) {
         //this function is called in the displayLetterHint method, line 276
-        //  String word = getCurrentCard().getAnswer();
+        //String word = getCurrentCard().getAnswer();
         StringBuilder string = new StringBuilder();
+        Random rng = new Random();
```

Comment:
Here, why use a simple use of random and not a random generator algorithm like Fisher-Yates shuffle? This can lead to collisions in a low percentage, but those cases can still happen, which are not taken account for in this scenario.

Reply:
I used a simple version of random because the feature should only have 2 options when clicked; either display a _ if the letter of the word is not shown yet or display the letter of the word in the correct spot. I used a simple random because it made more sense to implement into the code and it was personally easier to understand.